kafkaesk/consumer.py

```diff
@@ -425,25 +434,25 @@ async def healthy(self) -> None:
     # Event handlers
     async def on_partitions_revoked(self, revoked: typing.List[aiokafka.TopicPartition]) -> None:
         if revoked:
+            async with self._processing:
+                if self._auto_commit:
+                    await self._maybe_commit(forced=True)
+
             for tp in revoked:
+                del self._tp[tp]
                 CONSUMER_REBALANCED.labels(
                     partition=tp.partition,
                     group_id=self.group_id,
                     event="revoked",
                 ).inc()
             logger.info(f"Partitions revoked to {self}: {revoked}")
-            if not self._auto_commit:
-                return
-
-            # TODO: we dont want to wait forever for this lock, add a timeout and commit anyways.
-            async with self._processing:
-                await self._maybe_commit(forced=True)
 
     async def on_partitions_assigned(self, assigned: typing.List[aiokafka.TopicPartition]) -> None:
         if assigned:
             logger.info(f"Partitions assigned to {self}: {assigned}")
 
         for tp in assigned:
+            self._tp[tp] = None
```

Comment:
Do we need a `None` assignment here when you end up needing to filter it out when committing?

Reply:
totally agree!